### PR TITLE
[feat] Context API and Reducer

### PR DESCRIPTION
**Pull request by Hamdrive**
## Overview

Successfully implemented Context API and Reducer hook in the Pomodoro project, allowing global state management and use of reduce method to handle tasks

Features:
1. Use of context to access state throughout app, particularly in tasks and pomodoro page
2. Use of reducer to handle tasks

## Deployment

https://drivers-pomodoro-git-tasks-context-hamdrive.vercel.app/

### CODE REVIEW

File: src/context/tasks-context.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import { taskReducer } from "../utils";
+
+const tasksData = { tasks: [] };
+
+const TaskContext = createContext(null);
+
+const useTasks = () => useContext(TaskContext);
+
+const TaskProvider = ({ children }) => {
+  const [tasks, setTasks] = useState(
```
Comment (shivampandey0) [inline src/context/tasks-context.jsx]: This useState is not of use anymore here, as you are using useReducer here. You can handle all the things with the reducer only.

**Review by shivampandey0 — CHANGES_REQUESTED** [review]
Clean code practices followed.
- Added some suggestions

File: src/context/tasks-context.jsx (hunk 1 above)
Comment (Hamdrive) [reply src/context/tasks-context.jsx]: Yes I can do that, useReducer will handle all the same functionalities, 
will update code accordingly

**Review by shivampandey0 — APPROVED** [review]
Excellent use of useReducer to handle the tasks CRUD operations.
Good to merge

**Review by AnuragSahu11 — APPROVED** [review]
Nice styling, self explanatory function names, good to merge

**Comment by Hamdrive** [comment]
Successfully reviewed by 2 members, changes suggested to implementation of context and state, 
code has been updated and deployment is functional
Thus merging to dev an closing branch

**Pull request merged**